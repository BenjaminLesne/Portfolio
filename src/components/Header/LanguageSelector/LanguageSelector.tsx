import React from "react";
import { useEffect } from "react";
import UkFlag from "../../../assets/logos/languages/UkFlag.svg";
import FrenchFlag from "../../../assets/logos/languages/FrenchFlag.svg";
import { useTranslation } from "react-i18next";
import "./LanguageSelector.css";

export const LanguageSelector = () => {
  const { i18n } = useTranslation();
  const { changeLanguage, language } = i18n;
  const languageSelectorButton = document.querySelector(
    ".LanguageSelector__button"
  );

  function handleLanguageChange(language: string | null) {
    function updateToFrenchUI() {
      languageSelectorButton?.classList.remove("active");
      changeLanguage("fr");
    }

    function updateToEnglishUI() {
      languageSelectorButton?.classList.add("active");
      changeLanguage("en");
    }

    switch (language) {
      case "fr":
        if (languageSelectorButton?.classList.contains("active")) {
          updateToFrenchUI();
        }
        break;

      case "en":
        if (!languageSelectorButton?.classList.contains("active")) {
          updateToEnglishUI();
        }
        break;

      default:
        if (languageSelectorButton?.classList.contains("active")) {
          updateToFrenchUI();
        } else {
          updateToEnglishUI();
        }
        break;
    }
  }

  useEffect(() => {
    if (language === "en") {
      document
        ?.querySelector(".LanguageSelector__button")
        ?.classList.add("active");
    }
  }, []);

  return (
    <div className="LanguageSelector">
      <img
        className="LanguageSelector__flags"
        src={FrenchFlag}
        alt={language === "en" ? "French flag" : "Drapeau franÃ§ais"}
        onClick={() => handleLanguageChange("fr")}
        data-testid="LanguageSelectorButton--french"
      />
      <div
        data-testid="LanguageSelectorButton"
        className="LanguageSelector__button"
        onClick={() => handleLanguageChange(null)}
      >
        <div id="inner-circle" className="LanguageSelector__inner-circle"></div>
      </div>
      <img
        className="LanguageSelector__flags"
        src={UkFlag}
        alt={
          language === "en" ? "United Kingdom flag" : "Drapeau du Royaume-Uni"
        }
        onClick={() => handleLanguageChange("en")}
        data-testid="LanguageSelectorButton--english"
      />
    </div>
  );
};
