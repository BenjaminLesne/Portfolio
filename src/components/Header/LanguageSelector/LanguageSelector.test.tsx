import React from "react";
import { expect, test, describe } from "vitest";
import { render } from "@testing-library/react";
import { LanguageSelector } from "./LanguageSelector";

describe("The LanguageSelector component", () => {
  test("the snapshot match", () => {
    const languageSelector = render(<LanguageSelector />);
    expect(languageSelector).toMatchSnapshot();
  });
});
