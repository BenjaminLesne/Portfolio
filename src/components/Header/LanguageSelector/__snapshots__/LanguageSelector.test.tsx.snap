// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The LanguageSelector component > the snapshot match 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="LanguageSelector"
      >
        <img
          alt="Drapeau français"
          class="LanguageSelector__flags"
          data-testid="LanguageSelectorButton--french"
          src="/src/assets/logos/languages/FrenchFlag.svg"
        />
        <div
          class="LanguageSelector__button undefined"
          data-testid="LanguageSelectorButton"
        >
          <div
            class="LanguageSelector__inner-circle"
            id="inner-circle"
          />
        </div>
        <img
          alt="Drapeau du Royaume-Uni"
          class="LanguageSelector__flags"
          data-testid="LanguageSelectorButton--english"
          src="/src/assets/logos/languages/UkFlag.svg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="LanguageSelector"
    >
      <img
        alt="Drapeau français"
        class="LanguageSelector__flags"
        data-testid="LanguageSelectorButton--french"
        src="/src/assets/logos/languages/FrenchFlag.svg"
      />
      <div
        class="LanguageSelector__button undefined"
        data-testid="LanguageSelectorButton"
      >
        <div
          class="LanguageSelector__inner-circle"
          id="inner-circle"
        />
      </div>
      <img
        alt="Drapeau du Royaume-Uni"
        class="LanguageSelector__flags"
        data-testid="LanguageSelectorButton--english"
        src="/src/assets/logos/languages/UkFlag.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
